src/modules/_networking/virtual_network_gateway/virtual_network_gateway.tf

```diff
@@ -0,0 +1,21 @@
+resource "azurerm_virtual_network_gateway" "main" {
+  name                = var.settings.name
+  resource_group_name = local.resource_group_name
+  location            = local.location
+  generation          = var.settings.generation
+  type                = try(var.settings.type, "Vpn")
+  vpn_type            = try(var.settings.vpn_type, "RouteBased")
+  active_active       = try(var.settings.active_active, false)
+  enable_bgp          = try(var.settings.enable_bgp, false)
+  sku                 = try(var.settings.sku, "VpnGw1")
+
+  dynamic "ip_configuration" {
+    for_each = try(var.settings.ip_configurations, {})
+    content {
+      name                          = ip_configuration.value.name
+      public_ip_address_id          = can(ip_configuration.value.public_ip_address_id) || can(ip_configuration.value.public_ip_address_key) == false ? try(ip_configuration.value.public_ip_address_id, null) : var.resources.public_ips[ip_configuration.value.public_ip_address_ref].id
```

Comment:
1. These properties should be extracted to `_locals.tf`
2. Referencing the resource by id is a workaround in the original CAF, we should disallow it until required
3. Make sure you reference other resources with `_ref`